src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -369,20 +369,28 @@ public void CopyChangedValues(TStructuralType original)
 
                     if (deltaNestedResource.IsComplexType && newType != originalType)
                     {
-                        originalNestedResource = ReAssignComplexDerivedType(original, nestedResourceName, originalNestedResource, newType, originalType, deltaNestedResource.ExpectedClrType);
+                        originalNestedResource = ReAssignComplexDerivedType(originalNestedResource, newType, originalType, deltaNestedResource.ExpectedClrType);
+                        _structuredType.GetProperty(nestedResourceName).SetValue(original, (object)originalNestedResource);
                     }
 
                     deltaNestedResource.CopyChangedValues(originalNestedResource);
                 }
             }
+
+            return original;
```

Comment:
I'm sure I'm missing something obvious here, but I don't see the value in returning the parameter, especially because the generic type restriction for `TStructuralType` is that it is a `class`, so any modifications to it will be reflected on the underlying object anyway.

Reply:
Actually not really, and thats why we had to make this change, for complextype we are creating a new type like by doing original = new ...()